### PR TITLE
Initial set of changes for v21 release

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

Mods required for `v21` release:

- Change copy for new method of running local install, due to: https://github.com/vitessio/website/pull/1855. Change from having to add/replace hash of release commit, to specifying version for previous version's local install page.
- Use bot to create PRs that are generated by the tool, so that we only need two developers online for approving PRs.
The person running the releaser should create a Personal Access Token with `repo` rights in the Vitess repo. Define a environment variable called`VITESS_RELEASER_GH_TOKEN` set to the token.
- Remove -rc from target branch name for releases. All releases will be on the release-xx.0 branch, starting v21

### CODE REVIEW

File: go/releaser/utils/utils.go [hunk 1]
```diff
@@ -56,3 +56,16 @@ func ExecWithError(cmd string, args ...string) (string, error) {
...
+	if ghToken != "" {
+		currentGHToken := os.Getenv("GH_TOKEN")
+		os.Setenv("GH_TOKEN", ghToken)
+		fmt.Println("Setting GH_TOKEN to VITESS_RELEASER_GH_TOKEN")
+		return func() {
+			os.Setenv("GH_TOKEN", currentGHToken)
+		}
+	}
+	return func() {}
+}
```
Comment (frouioui) [inline go/releaser/utils/utils.go:71]: I was thinking we could rely on `gh auth switch`, that the user would have to be authenticated to the vitess-bot on their own. That way we don't have to change anything in the vitess-releaser. Although this is also good.

File: go/releaser/release/website_docs.go [hunk 2]
```diff
@@ -18,18 +18,21 @@ package release
...
 func WebsiteDocs(state *releaser.State) []string {
+	majorVersion, _ := strconv.Atoi(strings.Split(state.VitessRelease.MajorRelease, ".")[0])
+	prevVersion := majorVersion - 1
 	msg := []string{
 		"We want to open a Pull Request to update the documentation.",
 		"",
 		"There are several pages we want to update:",
 		"\t- https://vitess.io/docs/releases/: we must add the new release to the list with all its information and link.",
-		fmt.Sprintf("\t- https://vitess.io/docs/%s.0/get-started/local/#install-vitess: we must use '%s' for the 'version' variable, and the new SHA for the 'file' variable.", state.VitessRelease.MajorRelease, strings.ToLower(state.VitessRelease.Release)),
+		fmt.Sprintf("\t- Set https://vitess.io/docs/%d.0/get-started/local/#install-vitess: Set version='v%d.0' ", prevVersion, prevVersion),
```
Comment (frouioui) [inline go/releaser/release/website_docs.go:35]: I don't think we need to set anything manually, the `version` value is set automatically in the docs depending on the version of the docs the user is seeing.

Comment (rohit-nayak-ps) [reply go/releaser/release/website_docs.go]: Ah, had not understood that PR properly. Removed.

File: go/releaser/utils/utils.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/releaser/utils/utils.go:71]: When you mean `auth switch` you mean separately doing a `gh auth logout/login`? I don't see a `switch` subcommand. 

One problem with gh auth today is that it is global across all repos. So if we are switching between running the releaser and doing coding while waiting for a PR to merge, say, we would end up with the wrong gh user in the other application, right?

**Review by frouioui — APPROVED** [review]
Left a comment, otherwise looks good to me!

**Pull request merged**